lib/iex/lib/iex/introspection.ex

```diff
@@ -11,16 +11,22 @@ defmodule IEx.Introspection do
   Decomposes an introspection call into `{mod, fun, arity}`,
   `{mod, fun}` or `mod`.
   """
+  def decompose(atom, _context) when is_atom(atom), do: atom
+
+  def decompose({:__aliases__, _, _} = module, context) do
+    Macro.expand(module, context)
+  end
+
   def decompose({:/, _, [call, arity]} = term, context) do
     case Macro.decompose_call(call) do
       {_mod, :__info__, []} when arity == 1 ->
-        {:{}, [], [Module, :__info__, 1]}
+        {Module, :__info__, 1}
 
       {mod, fun, []} ->
-        {:{}, [], [mod, fun, arity]}
+        {Macro.expand(mod, context), fun, arity}
 
       {fun, []} ->
-        {:{}, [], [find_decompose_fun_arity(fun, arity, context), fun, arity]}
+        {find_decompose_fun_arity(fun, arity, context), fun, arity}
 
       _ ->
         term
```

Comment:
The error you are getting is because of this line here. We were previously returning the `term` as is, without escaping, and now we are escaping it always on `IEx.Helpers`. We also do it in the `decompose` clause below. My suggestion is to make it return `:error` instead.

Then, inside IEx.Helpers, you can write this helper:

```elixir
defp decompose(term, context) do
  case IEx.Introspection.decompose(term, context) do
    :error -> term
    m_mf_mfa -> Macro.escape(m_mf_mfa)
  end
end
```

And then, in `mix help`, you raise if `decompose` returns `:error`.

Reply:
Great idea! I just don't really know what to use on the raise message. Came up with `Invalid expression: #{module}` for now. Feel free to suggest something else!